api/server/middleware/user_agent.go

```diff
@@ -16,6 +17,7 @@ func NewUserAgentMiddleware(versionCheck string) Middleware {
 
 	return func(handler httputils.APIFunc) httputils.APIFunc {
 		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+			ctx = dockerversion.WithUserAgent(ctx, r.Header.Get("User-Agent"))
```

Comment:
@calavera The idea suggested by @aaronlehmann was to make all knowledge of the new type internal to the `dockerversion` package.  Doesn't this necessarily mean that accessor functions (which expose only built-in types like `string`) are required?  If so, how can I avoid importing `dockerversion` or some other package with the accessors?

The reason for defining a new type is to comply with the usage suggestions in this article:  https://blog.golang.org/context (under the section "Package userip").  Perhaps I'm misinterpreting that article?